mlflow/utils/autologging_utils/versioning.py

```diff
@@ -88,6 +85,7 @@ def is_flavor_supported_for_associated_package_versions(flavor_name):
     :return: True if the specified flavor is supported for the currently-installed versions of its
              associated packages
     """
+    module_version_info_dict = _load_version_file_as_dict()
```

Comment:
Lazily loading `module_version_info_dict` is good, but I am afraid it cannot address the demand https://github.com/mlflow/mlflow/pull/7550/files#r1051892533

Reply:
But if you hit some critical bug caused by current mlflow master implementation, we could reconsider it.